src/lib/db/access-store.ts

```diff
@@ -0,0 +1,163 @@
+import { EventEmitter } from 'events';
+import Knex from 'knex';
+import metricsHelper from '../metrics-helper';
+import { DB_TIME } from '../events';
+
+const T = {
+    ROLE_USER: 'role_user',
+    ROLES: 'roles',
+    ROLE_PERMISSION: 'role_permission',
+};
+
+export interface Permission {
```

Comment:
We haven't discussed this, but what do you think about a convention for naming interfaces? One quite common convention is to prefix interfaces with I. In this instance the interface would be IPermission. The benefit is that it clearly communicates in the code what it represents. Most of the time you can deduce it from the use context, but there's still some cognitive load involved.

Reply:
Let's try that. I will refactor all interfaces to "**I**MyInterface"